x-pack/plugins/log_explorer/public/components/log_explorer/types.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DataTableRecord } from '@kbn/discover-utils/types';
+
+export type RenderPreviousContent = () => React.ReactNode;
+
+export interface LogExplorerFlyoutContentProps {
+  doc: DataTableRecord;
+}
+
+export type FlyoutRenderContent = (
+  renderPreviousContent: RenderPreviousContent,
+  props: LogExplorerFlyoutContentProps
+) => React.ReactNode;
+
+export interface LogExplorerCustomizations {
+  flyout?: {
+    renderContent?: FlyoutRenderContent;
```

Comment:
Should we rename it to `customContent`. That would make it much more readable even in the place where we check and render it.

Reply:
I agreed with Felix when discussing the feature on this name because it felt was the most direct, also cause it tells about an action performed by the passed function (`render`), while `customContent` is more of a subject and I would expect to pass a primitive value or an object. It's mostly a matter of personal interpretation and reduces to naming, we can still change it in future if necessary.

Reply:
> also cause it tells about an action performed by the passed function (render)

good point, but as this is a render action performed on a Customisation. i feel may be `renderCustomContent` could be an option.

I am suggesting this from its usage perspective, where we have 2 render functions now,

`renderPreviousContent` and then `renderContent`

The actual content should be named as `renderContent` instead of `renderPreviousContent` as that's something originally present in the component. The customisation is actually the new content which should be conditionally rendered. 

So may be we call the `renderPreviousContent` as `renderContent`
and `renderContent` as `renderCustomContent`, wdyt ?